src/java.base/share/classes/java/lang/String.java

```diff
@@ -685,12 +685,6 @@ private String(Charset charset, byte[] bytes, int offset, int length) {
             cd.onMalformedInput(CodingErrorAction.REPLACE)
                     .onUnmappableCharacter(CodingErrorAction.REPLACE);
             char[] ca = new char[en];
-            if (charset.getClass().getClassLoader0() != null &&
-                    System.getSecurityManager() != null) {
-                bytes = Arrays.copyOfRange(bytes, offset, offset + length);
-                offset = 0;
```

Comment:
Was this defensive copy particularly important to security managers but has no effect for other scenarios?

Reply:
Application supplied Charsets are not any more or less trusted than any other class in the application.
So the copy is not (adequately) effective as a defensive measure here.